sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -344,8 +344,13 @@ case class MapFilter(
     Examples:
       > SELECT _FUNC_(array(1, 2, 3), x -> x % 2 == 1);
        [1,3]
+      > SELECT _FUNC_(array(0, 2, 3), (x, i) -> x > i);
```

Comment:
Here, the indices start at 0. but it sounds like the other built-in functions start at 1.

Reply:
I remember there was the (not-merged) PR to standardize one-based column indexes in built-in funcs: https://github.com/apache/spark/pull/24051
Better to fix them up for consistency?